src/main/java/com/redhat/rhjmc/containerjfr/net/SslConfiguration.java

```diff
@@ -145,74 +118,117 @@ Path discoverKeyStorePathInDefaultLocations() {
 
     Pair<Path, Path> discoverKeyCertPathPairInDefaultLocations() {
         Path home = fs.pathOf(System.getProperty("user.home", "/"));
-        Pair<Path, Path> ret = new Pair<>(null, null);
+        Path key = null;
+        Path cert = null;
 
         // discover keyPath
         if (fs.exists(home.resolve("container-jfr-key.pem"))) {
-            ret.left = home.resolve("container-jfr-key.pem");
-        } else {
-            logger.warn("unable to locate container-jfr-key.pem");
+            key = home.resolve("container-jfr-key.pem");
         }
 
         // discover certPath
         if (fs.exists(home.resolve("container-jfr-cert.pem"))) {
-            ret.right = home.resolve("container-jfr-cert.pem");
-        } else {
-            logger.warn("unable to locate container-jfr-cert.pem");
+            cert = home.resolve("container-jfr-cert.pem");
         }
 
-        if (ret.left == null || ret.right == null) {
+        if (key == null || cert == null) {
             return null;
         }
 
-        return ret;
+        return Pair.of(key, cert);
+    }
+
+    HttpServerOptions applyToHttpServerOptions(HttpServerOptions options) {
+        return strategy.applyKeyStoreOptions(options);
+    }
+
+    boolean enabled() {
+        return strategy.enabled();
+    }
+
+    SslConfigurationStrategy keyStoreStrategy(Path path, String password) throws SslConfigurationException {
+        return new KeyStoreStrategy(path, password);
     }
 
-    HttpServerOptions setToHttpServerOptions(HttpServerOptions options) {
-        if (!enabled()) {
+    SslConfigurationStrategy keyCertStrategy(Path keyPath, Path certPath) throws SslConfigurationException {
+        return new KeyCertStrategy(keyPath, certPath);
+    }
+    
+    SslConfigurationStrategy noSslStrategy() {
+        return new NoSslStrategy();
+    }
+
+    interface SslConfigurationStrategy {
+        HttpServerOptions applyKeyStoreOptions(HttpServerOptions options);
+        
+        default boolean enabled() {
+            return !(this instanceof NoSslStrategy);
```

Comment:
I think this is a little bit *too* clever :-). Rather than defaulting to an `instanceof` check, how about simply defaulting to return `true`, and then the `NoSslStrategy` can override that to return `false`?

Reply:
Yes, that sounds better. I'll update as suggested.